library/core/src/cell.rs

```diff
@@ -1569,20 +1572,65 @@ impl<T: ?Sized + fmt::Display> fmt::Display for RefMut<'_, T> {
 ///
 /// Note that while mutating or mutably aliasing the contents of an `&UnsafeCell<T>` is
 /// ok (provided you enforce the invariants some other way), it is still undefined behavior
-/// to have multiple `&mut UnsafeCell<T>` aliases.
+/// to have multiple `&mut UnsafeCell<T>` aliases. That is, `UnsafeCell` is a wrapper
+/// designed to have a special interaction (_w.r.t._ the compiler) with _shared_
```

Comment:
The key point is interaction with the language (specification), the compiler is just an implementation of said specification.